Testscripts/Windows/VERIFY-SRIOV-FAILSAFE-XDP.ps1

```diff
@@ -0,0 +1,196 @@
+        -command $ping_command -RunInBackground -runAsSudo
+
+    # Start XDPDump
+    $xdp_command = "cd /root/bpf-samples/xdpdump && timeout 10 ./xdpdump -i $NIC > ~/xdpdumpout$LogSuffix.txt"
+    Write-LogInfo "Starting command: $xdp_command on $($VMData.RoleName)"
+    $testJob = Run-LinuxCmd -ip $VMData.PublicIP -port $VMData.SSHPort -username $user -password $password `
+        -command $xdp_command -RunInBackground -runAsSudo
+    $timer = 0
+    while ((Get-Job -Id $testJob).State -eq "Running" ) {
+        $currentStatus = Run-LinuxCmd -ip $VMData.PublicIP -port $VMData.SSHPort -username $user -password $password `
+            -command "tail -2 ~/xdpdumpout$LogSuffix.txt | head -1" -runAsSudo
+        Write-LogInfo "Current Test Status: $currentStatus"
+        Wait-Time -seconds 5
+        $timer += 1
+        if ($timer -gt 15) {
+            Throw "XDPSetup did not stop after 5 mins. Please check logs."
+        }
+    }
+
+    $currentStatus = Run-LinuxCmd -ip $VMData.PublicIP -port $VMData.SSHPort -username $user -password $password `
+        -command "tail -1 ~/xdpdumpout$LogSuffix.txt | head -1" -runAsSudo
+    if ( ($currentStatus -inotmatch "unloading xdp") ) {
+        Write-LogErr "Test Aborted. Last known status : $currentStatus."
+        Throw "XDP Execution failed"
+    }
+    Write-LogInfo "XDPDump application ran successfully."
+    return "PASS"
+}
+
+function Main {
+    try{
+        $noClient = $true
+		$noServer = $true
+		foreach ($vmData in $allVMData) {
+			if ($vmData.RoleName -imatch "receiver") {
+				$clientVMData = $vmData
+				$noClient = $false
+			}
+			elseif ($vmData.RoleName -imatch "sender") {
+				$noServer = $false
+				$serverVMData = $vmData
+			} else {
+				Write-LogErr "VM role name is not matched with sender or receiver"
+			}
+		}
+		if ($noClient) {
+			Throw "No any receiver VM defined. Aborting Test."
+		}
+		if ($noServer) {
+			Throw "No any sender VM defined. Aborting Test."
+        }
+
+        # CONFIGURE VM Details
+        Write-LogInfo "CLIENT VM details :"
+        Write-LogInfo "  RoleName : $($clientVMData.RoleName)"
+        Write-LogInfo "  Public IP : $($clientVMData.PublicIP)"
+        Write-LogInfo "  SSH Port : $($clientVMData.SSHPort)"
+        Write-LogInfo "  Internal IP : $($clientVMData.InternalIP)"
+        Write-LogInfo "SERVER VM details :"
+        Write-LogInfo "  RoleName : $($serverVMData.RoleName)"
+        Write-LogInfo "  Public IP : $($serverVMData.PublicIP)"
+        Write-LogInfo "  SSH Port : $($serverVMData.SSHPort)"
+        Write-LogInfo "  Internal IP : $($serverVMData.InternalIP)"
+
+        # Check for compatible kernel
+        $currentKernelVersion = Run-LinuxCmd -ip $clientVMData.PublicIP -port $clientVMData.SSHPort `
+            -username $user -password $password -command "uname -r"
+        if ((Compare-KernelVersion $currentKernelVersion $MIN_KERNEL_VERSION) -lt 0 -or $global:DetectedDistro -ne "UBUNTU"){

```

Comment:
I know this is for the future kernel version, but you are not going to check with a new kernel build step? Then how did you get the change base test result passed?

Reply:
I didn't understand your ask completely. But if you are wondering about kernel version used for testing is 5.7 as I started this test with linux-next kernel.

Reply:
This test will be skipped until the kernel version is 5.7 in our daily work. Correct?

Reply:
*5.6
Yes

Reply:
Updated Log msg with minimum kernel version required. Also, added a comment for the future work.